### PR TITLE
feat: added native slack webhook type

### PR DESCRIPTION
**Pull request by sheensantoscapadngan**
# Description 📣
- This PR adds slack type to project webhooks
- This PR also modifies the secret path input in add webhook form to use latest SecretPathInput component for autocomplete support

image
image
image

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Comment by sheensantoscapadngan** [comment]
TO DO: 

add documentation

**Comment by sheensantoscapadngan** [comment]
image

**Comment by sheensantoscapadngan** [comment]
Added the folllowing to do the docs:
image

File: backend/src/db/migrations/20240702055253_add-encrypted-webhook-url.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import { TableName } from "../schemas";
+
+export async function up(knex: Knex): Promise<void> {
+  const hasEncryptedURL = await knex.schema.hasColumn(TableName.Webhook, "encryptedUrl");
+  const hasUrlIV = await knex.schema.hasColumn(TableName.Webhook, "urlIV");
+  const hasUrlTag = await knex.schema.hasColumn(TableName.Webhook, "urlTag");
+  const hasType = await knex.schema.hasColumn(TableName.Webhook, "type");
+
+  await knex.schema.alterTable(TableName.Webhook, (tb) => {
```
Comment (akhilmhdh) [inline backend/src/db/migrations/20240702055253_add-encrypted-webhook-url.ts:13]: Check for table exist or not is missing

File: backend/src/db/schemas/webhooks.ts [hunk 2]
```diff
@@ -21,7 +21,11 @@ export const WebhooksSchema = z.object({
   keyEncoding: z.string().nullable().optional(),
   createdAt: z.date(),
   updatedAt: z.date(),
-  envId: z.string().uuid()
+  envId: z.string().uuid(),
+  encryptedUrl: z.string().nullable().optional(),
```
Comment (akhilmhdh) [inline backend/src/db/schemas/webhooks.ts:25]: May be keep the naming convention same by as `urlCipherText` or changing test as `encrypted` prepended

File: backend/src/services/webhook/webhook-fns.ts [hunk 3]
```diff
@@ -12,63 +12,131 @@ import { logger } from "@app/lib/logger";
...
-    if (rootEncryptionKey && keyEncoding === SecretKeyEncoding.BASE64) {
-      // case: encoding scheme is base64
-      secretKey = decryptSymmetric({
+  const encryptionKey = appCfg.ENCRYPTION_KEY;
+  const rootEncryptionKey = appCfg.ROOT_ENCRYPTION_KEY;
+
+  let decryptedSecretKey = "";
+  let decryptedUrl = url;
+
+  if (rootEncryptionKey && keyEncoding === SecretKeyEncoding.BASE64) {
```
Comment (akhilmhdh) [inline backend/src/services/webhook/webhook-fns.ts:29]: I know this was written like this with if before you started working on it.

Can you please switch this if to our reusable function `infisicalSymmetricEncrypt` and `infisicalSymmetricDecrypt`.

**Review by akhilmhdh — CHANGES_REQUESTED** [review]
Small changes

**Comment by akhilmhdh** [comment]
Also one more reminder.

Try a fake slack so that it fails and check the log if the url is getting printed. If so redact it for security reasons

**Comment by akhilmhdh** [comment]
Screenshot 2024-07-04 at 1 04 10 AM


Can we also make the environment in new line

File: backend/src/services/webhook/webhook-fns.ts (hunk 3 above)
Comment (sheensantoscapadngan) [reply backend/src/services/webhook/webhook-fns.ts:29]: yeah sure will do

File: backend/src/db/migrations/20240702055253_add-encrypted-webhook-url.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { Knex } from "knex";
```
Comment (sheensantoscapadngan) [inline backend/src/db/migrations/20240702055253_add-encrypted-webhook-url.ts:1]: updated slack message layout @akhilmhdh 
image

**Comment by sheensantoscapadngan** [comment]
> Also one more reminder.
> 
> Try a fake slack so that it fails and check the log if the url is getting printed. If so redact it for security reasons

checked this one as well @akhilmhdh. We're not logging the URL in the case of errors

**Review by akhilmhdh — APPROVED** [review]

**Pull request merged**